sumomongodbatlascollector/api.py

```diff
@@ -210,68 +207,56 @@ def build_send_params(self):
-        data_availablity_max_endDate = int(get_current_timestamp() - 5*60)
+        data_availablity_max_endDate = int(get_current_timestamp() - 5 * 60)
         api_endDate = kwargs['params']["endDate"]
         if api_endDate < data_availablity_max_endDate:
             return True, {"last_time_epoch": api_endDate}
         else:
             return False, {}
 
     def transform_data(self, content):
-        # assuming file content is small so inmemory possible
-        # https://stackoverflow.com/questions/11914472/stringio-in-python3
-        # https://stackoverflow.com/questions/8858414/using-python-how-do-you-untar-purely-in-memory
         all_logs = []
         last_time_epoch = self.DEFAULT_START_TIME_EPOCH
         results = gzip.GzipFile(fileobj=BytesIO(content))
         last_line = ""
-        for line_no, line in enumerate(results.readlines()):
-            if not line.strip():
-                # for JSONDecoderror in case of empty lines
+
+        for line in results:
+            line = line.strip()
+            if not line:
                 continue
+
             line = line.decode('utf-8')
             hostname_alias = self._replace_cluster_name(self.hostname, self.cluster_mapping)
             cluster_name = self._get_cluster_name(hostname_alias)
 
             if "audit" in self.filename:
-                if last_line:
-                    line = last_line + line
+                line = last_line + line
                 try:
                     msg = json.loads(line)
                     last_line = ""
-                except ValueError as e:
-                    # checking for multiline messages
+                except ValueError:
                     last_line = line
-                    self.log.warn("Multiline Message in line no: %d last_log: %s current_log: %s" % (line_no, all_logs[-1:], line))
                     continue
-                msg['project_id'] = self.api_config['PROJECT_ID']
-                msg['hostname'] = hostname_alias
-                msg['cluster_name'] = cluster_name
-                current_date = msg['ts']['$date']
             else:
-                if last_line:
-                    line = last_line + line
+                line = last_line + line
                 try:
                     msg = json.loads(line)
                     last_line = ""
-                except ValueError as e:
-                    # checking for multiline messages
+                except ValueError:
                     last_line = line
-                    self.log.warn("Multiline Message in line no: %d last_log: %s current_log: %s" % (line_no, all_logs[-1:], line))
                     continue
-                msg['project_id'] = self.api_config['PROJECT_ID']
-                msg['hostname'] = hostname_alias
-                msg['cluster_name'] = cluster_name
-                current_date = msg['t']['$date']
 
+            msg['project_id'] = self.api_config['PROJECT_ID']
```

Comment:
revert back this change, the older if conditions were there for different mongodb versions which have different log format.